src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs

```diff
@@ -143,7 +143,14 @@ public bool IsNameCompliant(string name, out string failureReason)
                 return true;
             }
 
-            var spanToCheck = TextSpan.FromBounds(Prefix.Length, name.Length - Suffix.Length);
+            name = StripCommonPrefixes(name, out var prefix);
+            if (prefix != string.Empty)
+            {
+                failureReason = string.Format(WorkspacesResources.Incorrect_prefix);
```

Comment:
in what contexts is this string shown to the user?

Reply:
If the name begins with the specified prefix (prefix can be "") but has one or more additional common prefixes (_, _m, _s, _t) this will show in the error list.